src/main/java/com/walmartlabs/x12/standard/StandardX12Parser.java

```diff
@@ -57,6 +58,8 @@
 
     public static final String TRANSACTION_HEADER_ID = "ST";
     public static final String TRANSACTION_TRAILER_ID = "SE";
+    
+    private TransactionSetParser transactionParser;
```

Comment:
Assuming we have two versions, one that accepts a single `TransactionSetParser` and another that takes a `List< TransactionSetParser>`, with the former would we throw an exception if it was null to stop the object from being created? I am not sure I like that.

With the latter, if we take a list there would not be an easy way to let the consumer/user know whether all of the parsers were added or not.

thoughts?

Reply:
How would `registerTransactionSetParser()` let the consumer know if it failed? It has a void return right now

Reply:
I feel that exposing `registerTransactionSetParser` method allows change in behavior runtime

Reply:
the `registerTransactionSetParser` methods both return boolean

Reply:
> I feel that exposing registerTransactionSetParser method allows change in behavior runtime after object is created

That is possible. This is basically the setter vs constructor discussion that often came up w/ Spring IoC, especially before `@Autowired`. 

Also that would be a different conversation than just adding constructors and dealing w/ how they should act. You would really be suggesting we remove the register methods. 😄

Reply:
Even w/ a constructor, the consumer has access to the `TransactionSetParser` object reference (because they create and pass in the objects either through the Collection or directly depending on the version of constructor we go with). Thus technically the object behavior can be altered regardless of the approach. 

I am going to leave the `registerTransactionSetParser` methods, but am open to constructor if we can decide how they should behave under certain conditions.